ocaml/sdk-gen/go/gen_go_helper.ml

```diff
@@ -169,32 +195,230 @@ module Json = struct
     | _ ->
         [("event", `Null); ("session", `Null)]
 
+  let of_result obj msg =
+    match msg.msg_result with
+    | None ->
+        `Null
+    | Some (t, _d) ->
+        if obj.name = "event" && String.lowercase_ascii msg.msg_name = "from"
+        then
+          `O
+            [
+              ("type", `String "EventBatch")
+            ; ("func_partial_type", `String "EventBatch")
+            ]
+        else
+          let t', _ = string_of_ty_with_enums t in
+          `O
+            [
+              ("type", `String t')
+            ; ("func_partial_type", `String (get_fp_type t))
+            ]
+
+  let of_param class_name params =
```

Comment:
If the `session` could be handled in a dedicated template, this function could be as simple as the following?
```
let of_param param =
  let internal_name name =
    let name = name |> snake_to_camel |> String.uncapitalize_ascii in
    (* I can't find any usage on 'typeKey' *)
    match name with "interface" -> "iface" | _ -> name
  in
  `O
  [
  ; ("type", `String (string_of_ty_with_enums param |> fst))
  ; ("name", `String param.param_name)
  ; ("internal_name", `String (internal_name param.param_name))
  ; ("doc", `String param.param_doc)
  ; ("func_name_suffix", `String (func_name_suffix param.param_name))
  ]

`A (List.map of_param params)
```

Reply:
Here we need to replace "type" with "typeKey" for type is a key in Go.